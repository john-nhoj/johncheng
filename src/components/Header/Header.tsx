import React, { useState } from 'react';
import Image from 'next/image';
import classNames from 'classnames';
type HeaderLinkProps = {
  label: string;
  link: string;
  className?: string;
};

const HeaderLink = ({ label, link, className }: HeaderLinkProps) => (
  <a
    href={link}
    className={classNames(
      className,
      'flex',
      'items-center',
      'first:ml-0',
      'ml-8',
      'header-link'
    )}
  >
    {label}
  </a>
);

export const Header = () => {
  const [menuOpen, toggleMenu] = useState(false);

  return (
    <header
      className={classNames(
        'bg-white',
        'sm:px-20',
        'md:px-40',
        'xl:px-96',
        'flex',
        'flex-col',
        'md:flex-row',
        'border-solid',
        'border-b-2',
        'border-border'
      )}
    >
      <div className="flex justify-between">
        <a
          href="/"
          className={classNames('flex', 'items-center', 'p-2', 'md:p-4')}
        >
          <Image src="/logo.svg" alt="Website logo" width={200} height={28} />
        </a>
        <button
          className="flex items-center p-2 md:hidden"
          onClick={() => toggleMenu(!menuOpen)}
        >
          <Image
            src="/menu.svg"
            alt="Menu toggle"
            width={25}
            height={25}
            className="md:hidden"
          />
        </button>
      </div>
      <div
        className={classNames(
          'flex',
          'md:flex',
          'flex-1',
          'items-center',
          'flex-row',
          'justify-end',
          'p-2',
          {
            hidden: !menuOpen,
          }
        )}
      >
        <HeaderLink label="Home" link="/" />
        <HeaderLink label="About" link="/about" />
        <HeaderLink label="Blog" link="/blog" />
      </div>
    </header>
  );
};
