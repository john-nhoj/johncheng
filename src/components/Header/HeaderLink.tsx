import React from 'react';
import classNames from 'classnames';

type HeaderLinkProps = {
  label: string;
  link: string;
  className?: string;
};

export const HeaderLink = ({ label, link, className }: HeaderLinkProps) => (
  <a
    href={link}
    className={classNames(
      className,
      'flex',
      'items-center',
      'first:ml-0',
      'ml-8',
      'header-link'
    )}
  >
    {label}
  </a>
);
