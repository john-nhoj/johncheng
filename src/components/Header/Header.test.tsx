import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from './Header';

test('Header', () => {
  const header = renderer.create(<Header />);
  let tree = header.toJSON();
  expect(tree).toMatchSnapshot();
});
