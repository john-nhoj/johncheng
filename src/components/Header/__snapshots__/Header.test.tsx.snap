// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<header
  className="bg-white border-solid border-b-2 border-border"
>
  <div
    className="flex flex-col md:flex-row layout__margin--horizontal"
  >
    <div
      className="flex justify-between"
    >
      <a
        className="flex items-center layout__padding--full"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDUiIGhlaWdodD0iNDUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <noscript>
            <img
              alt="Website logo"
              decoding="async"
              src="/logo.svg"
              srcSet="/_next/image?url=%2Flogo.svg&w=48&q=75 1x, /_next/image?url=%2Flogo.svg&w=96&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
          <img
            alt="Website logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
      </a>
      <button
        className="flex items-center p-2 md:hidden"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <noscript>
            <img
              alt="Menu toggle"
              className="md:hidden"
              decoding="async"
              src="/menu.svg"
              srcSet="/_next/image?url=%2Fmenu.svg&w=32&q=75 1x, /_next/image?url=%2Fmenu.svg&w=64&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
          <img
            alt="Menu toggle"
            className="md:hidden"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
      </button>
    </div>
    <div
      className="flex md:flex flex-1 items-center flex-row justify-end hidden"
    >
      <a
        className="flex items-center first:ml-0 ml-2 layout__padding--full header-link"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </a>
      <a
        className="flex items-center first:ml-0 ml-2 layout__padding--full header-link"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        About
      </a>
      <a
        className="flex items-center first:ml-0 ml-2 layout__padding--full header-link"
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
    </div>
  </div>
</header>
`;
